src/main/java/com/liveramp/hyperminhash/HyperMinHash.java

```diff
@@ -40,22 +33,26 @@ public HyperMinHash(int p, int r) {
   }
 
   static HyperMinHash wrapRegisters(int p, int r, long[] registers) {
-   return new HyperMinHash(p, r, registers);
+    return new HyperMinHash(p, r, registers);
   }
 
   private HyperMinHash(int p, int r, long[] registers) {
     // Ensure that the number of registers isn't larger than the largest array java can hold in memory
     // biggest java array can be of size Integer.MAX_VALUE
-    Preconditions.checkArgument(p > 1 && p < 31);
+    if (!(p > 1 && p < 31)) {
+      throw new IllegalArgumentException("precision (p) must be between 1 and 31.");
```

Comment:
I think we may want to enforce that this is at least 4, since the bias-correction doesn't work at lower precisions.